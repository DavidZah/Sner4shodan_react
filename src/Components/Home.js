import React from 'react';
import Services from './Service';
import {getJsonData} from './../Api';

export function Home() {
    const projectName = 'Sner4Shodan';
    const json_data = getJsonData();
    return (
        <div>
            <div className="projectName">{projectName}</div>
            {json_data.data.map((service, serviceKey) => (
                <div key={serviceKey}>
                    <h1>{service.host_name}</h1>
                    <Services renderOneService={true} serviceValue={serviceKey} />
                </div>
            ))}
        </div>
    );
}
