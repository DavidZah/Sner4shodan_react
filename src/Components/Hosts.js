import React from 'react';
import {getJsonData} from './../Api'
import { Route, Redirect } from 'react-router'


 function HostButton(props) {
    return (
        <div>
            <button className="hostButton" onClick={() => props.onClick()}>
                {props.host_name}
            </button>
        </div>
    );
}

export class Hosts extends React.Component{
    
    handleClick(i) {

    }

    renderAll() {
        const json_data = getJsonData();
        let data = []; 
        for (let i in json_data.data) {
            data.push(<HostButton host_name={json_data.data[i].host_name} onClick={() => this.handleClick(i)} />);
        }
        return data;
    }

    render(){
        return(
            this.renderAll()
        );
    }
}