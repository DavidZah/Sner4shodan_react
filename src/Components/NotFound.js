function BackButton() {
    return (
        <button className="back" onClick={() => this.props.onClick()}>
            Back
        </button>
    );
}


class ErrorNotFound extends React.Component {


render(){
  return(
      <div>
      <h3>Not found </h3>
      <BackButton 
      
      />
    </div>
  );
}
}

export default Host;
