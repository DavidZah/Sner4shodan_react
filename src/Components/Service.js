import React from 'react';

const json_data =  {data:[{
    host_name: "Microsoft",
    ip: "192.168.154.2",
    "servic": [{
        name: 'Samba',
        port: '666',
        type: 'TCP'
    },
    {
        name: 'HTTP',
        port: '80',
        type: 'UDP'
    },
    {
        name: 'SSH',
        port: '23',
        type: 'TEST1'
    }
    ]
 },
 {
  host_name: "Sony",
    ip: "192.136.645.8",
    "servic": [{
        name: 'Telnet',
        port: '48',
        type: 'TCP'
    },
    {
        name: 'HTTPS',
        port: '443',
        type: 'UDP'
    },
    {
        name: 'SSH',
        port: '23',
        type: 'TCP'
    }
    ]
 },
 {
    host_name: "Pentagon",
    ip: "88.66.154.9",
    "servic": [{
        name: 'UNKOWN',
        port: '2864',
        type: 'TCP'
    },
    {
        name: 'HTTP',
        port: '80',
        type: 'UDP'
    },
    {
        name: 'SSH',
        port: '23',
        type: 'TEST1'
    },
    {
        port: '6500',
        type: 'tcp'
    },
    {
        name: 'test',
        port: '80',
        type: 'tcp'
    }
    ]
 }]
 };


class Services extends React.Component{
    
    renderAll(){
       let data =[]; 
        for(let x in json_data.data){
            for(let i in json_data.data[x].servic){
                data.push(<Service
                serviceName = {json_data.data[x].servic[i].name}
                port = {json_data.data[x].servic[i].port}
                type = {json_data.data[x].servic[i].type}
                />)
            }
        }
        return(
            data
        );
   }
   renderService(x){
       let data =[]; 
    for(let i in json_data.data[x].servic){ 
        data.push(<Service
        serviceName = {json_data.data[x].servic[i].name}
        port = {json_data.data[x].servic[i].port}
        type = {json_data.data[x].servic[i].type}
        />);
    }
        return(
            data
        );   
    }


    render(){
       if(this.props.renderOneService){
            return(
                this.renderService(this.props.serviceValue)
            );
        }
        return(
            this.renderAll()
        );
    }
}


function Service(props){
    return(
        <div>
        <span className="service">Name:{props.serviceName} </span>
        <span className="port">Port:{props.port} </span>
        <span className="type">Type:{props.type} </span>
        </div>        
    );
}
export default Services;