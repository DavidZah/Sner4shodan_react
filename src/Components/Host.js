import React from 'react';
import { getJsonData } from './../Api';
import { ErrorNotFound } from './NotFound';
import { Services } from './Service';
import { Redirect } from 'react-router';

function BackButton(props) {
    return (
        <button class="btn btn-default" onClick={() => props.onClick()}>
            Back
        </button>
    );
}

export class Host extends React.Component {
    constructor(props) {
        super(props);
        this.state = { returnToHost: false };
    }

    findId(json_data) {
        var id_name = this.props.match.params.id;

        for (var i in json_data.data) {
            if (json_data.data[i].host_name === id_name) {
                return json_data.data[i];
            }
        }
        return false;
    }

    handleClick() {
        this.setState(() => ({
            returnToHost: true,
        }));
    }

    render() {
        const json_data = getJsonData();
        var x = this.findId(json_data);

        if (this.state.returnToHost) {
            this.setState(() => ({
                returnToHost: false,
            }));
            return <Redirect to="/hosts" />;
        }

        if (x !== false) {
            return (
                <div>
                    <Services renderOneService={true} service={x} />
                    <BackButton onClick={() => this.handleClick()} />
                </div>
            );
        }

        return <ErrorNotFound />;
    }
}
