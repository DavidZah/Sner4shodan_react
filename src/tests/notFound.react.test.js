import React from 'react';  
import renderer from 'react-test-renderer';  
import { ErrorNotFound } from '../Components/NotFound';


it('renders correctly', () => {  
const tree = renderer.create(<ErrorNotFound/>).toJSON();  
  expect(tree).toMatchSnapshot();  
});

