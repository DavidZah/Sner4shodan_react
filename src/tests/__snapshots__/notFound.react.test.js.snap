// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3>
    Not found 
  </h3>
  <button
    className="btn btn-default navbar-btn"
    onClick={[Function]}
  >
    Back
  </button>
</div>
`;
